internal/controller/services/secretgenerator/secret_test.go

```diff
@@ -14,7 +18,85 @@ const (
 	errUnsupportedType        = "secret type is not supported"
 )
 
+var (
+	base64Regex = regexp.MustCompile("^[A-Za-z0-9+/]*={0,3}$")
+)
+
 func TestNewSecret(t *testing.T) {
+	tests := []struct {
+		name               string
+		secretName         string
+		secretType         string
+		complexity         int
+		expectedResult     string
+		expectedErrMessage string
+	}{
+		{
+			name:           "random case",
+			secretName:     "my-secret",
+			secretType:     "random",
+			complexity:     1,
+			expectedResult: "success",
+		},
+		{
+			name:           "oauth case",
+			secretName:     "another-secret",
+			secretType:     "oauth",
+			complexity:     1,
+			expectedResult: "success",
+		},
+		{
+			name:               "unsupported secret type",
+			secretName:         "my-secret",
+			secretType:         "·%$@&?",
+			complexity:         1,
+			expectedResult:     "error",
+			expectedErrMessage: errUnsupportedType,
+		},
+		{
+			name:           "zero complexity",
+			secretName:     "my-secret",
+			secretType:     "random",
+			complexity:     0,
+			expectedResult: "nil",
+		},
+		{
+			name:           "empty name",
+			secretName:     "",
+			secretType:     "random",
+			complexity:     1,
+			expectedResult: "success",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			secret, err := secretgenerator.NewSecret(tt.secretName, tt.secretType, tt.complexity)
+			switch tt.expectedResult {
+			case "error":
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErrMessage)
+				return // Early return after error validation
+			case "nil":
+				require.NoError(t, err)
```

Comment:
[require.Nil()](https://pkg.go.dev/github.com/stretchr/testify/require#Nil) could be added here.
```suggestion
				require.NoError(t, err)
				require.Nil(t, secret)
```

Reply:
```
=== RUN   TestNewSecret/zero_complexity
    secret_test.go:80: 
        	Error Trace:	/Users/asanzgom/work/opendatahub-operator/internal/controller/services/secretgenerator/secret_test.go:80
        	Error:      	Expected nil, but got: &secretgenerator.Secret{Name:"my-secret", Type:"random", Complexity:0, Value:"", OAuthClientRoute:""}
        	Test:       	TestNewSecret/zero_complexity
```

Reply:
Then I would change the test as the expected result is not nil but a normal secret, just with empty value. So expectedResult can be left nil but the secret should be checked that it has a zero value.

Reply:
```
=== RUN   TestNewSecret/zero_complexity
    secret_test.go:88: 
        	Error Trace:	/Users/asanzgom/work/opendatahub-operator/internal/controller/services/secretgenerator/secret_test.go:88
        	Error:      	Should NOT be empty, but was 
        	Test:       	TestNewSecret/zero_complexity
```

Reply:
@MarianMacik I remove the NotEmpty assert from the success case then

Reply:
@asanzgom I would revert that nil change and leave it there. Then based on the method the value will be empty, you are right because when slice is converted to string here
https://github.com/opendatahub-io/opendatahub-operator/blob/135e2bab1b8c91a84444ee6f79af5f4badb78590/internal/controller/services/secretgenerator/secret.go#L100

it contains non-printable characters (ordinal code 0), so it will be empty basically. So as I said, the secret is there, just its value is "zeroed", i.e. is empty.